### PR TITLE
fix: detangle and unify flow initialization, fix netToGross bug

### PR DESCRIPTION
**Pull request by paveltomin**

### CODE REVIEW

File: src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp
Comment (paveltomin) [inline src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp:539]: this double update is still confusing but i don't really want to touch the constitutives...

**Comment by paveltomin** [comment]
one case requires rebaseline:
```
Rank 0 is comparing /tmp/tmp.MtQrIy5KpC/GEOS_integratedTests_working/poromechanics/PoroElastic_hybridHexPrism_co2_sequential_3d_01/0to20_restart_000000020/rank_0000000.hdf5 with /tmp/tmp.MtQrIy5KpC/GEOS_integratedTests_baselines/poromechanics/PoroElastic_hybridHexPrism_co2_sequential_3d_01/0to20_restart_000000020/rank_0000000.hdf5 
********************************************************************************
Error: /Problem/Solvers/multiphasePoroelasticity/SolverStatistics/numSuccessfulNonlinearIterations/__values__
	Scalar values of types int32 and int32 differ: 58, 60.
********************************************************************************
********************************************************************************
Error: /Problem/Solvers/linearElasticity/SolverStatistics/numSuccessfulLinearIterations/__values__
	Scalar values of types int32 and int32 differ: 58, 60.
********************************************************************************
********************************************************************************
Error: /Problem/Solvers/linearElasticity/SolverStatistics/numSuccessfulNonlinearIterations/__values__
	Scalar values of types int32 and int32 differ: 58, 60.
********************************************************************************
********************************************************************************
Error: /Problem/Solvers/linearElasticity/maxForce/__values__
	Scalar values of types float64 and float64 differ: 59056644.21638392, 59057351.806163654.
********************************************************************************
********************************************************************************
Error: /Problem/Solvers/twoPhaseFlow/SolverStatistics/numSuccessfulLinearIterations/__values__
	Scalar values of types int32 and int32 differ: 122, 130.
********************************************************************************
********************************************************************************
Error: /Problem/Solvers/twoPhaseFlow/SolverStatistics/numSuccessfulNonlinearIterations/__values__
	Scalar values of types int32 and int32 differ: 122, 130.
********************************************************************************
********************************************************************************
Error: /Problem/domain/MeshBodies/mesh/meshLevels/Level0/nodeManager/incrementalDisplacement
	Arrays of types float64 and float64 have 10164 values of which 308 fail both the relative and absolute tests.
		Max absolute difference is at index (np.int64(3367), np.int64(2)): value = 3.329290118136317e-05, base_value = 2.0853878413888033e-05
		Max relative difference is at index (np.int64(31), np.int64(1)): value = 2.834348701588979e-18, base_value = -1.1883888138405466e-20
	Statistics of the q values greater than 1.0 defined by absolute tolerance: N = 308
		max = 1.2439022767475134, mean = 0.03769400838628123, std = 0.2132295115203364
		max is at index (np.int64(3367), np.int64(2)), value = 3.329290118136317e-05, base_value = 2.0853878413888033e-05
	Statistics of the q values greater than 1.0 defined by relative tolerance: N = 0
********************************************************************************
********************************************************************************
Error: /Problem/domain/MeshBodies/mesh/meshLevels/Level0/nodeManager/totalDisplacement
	Arrays of types float64 and float64 have 10164 values of which 1232 fail both the relative and absolute tests.
		Max absolute difference is at index (np.int64(3337), np.int64(2)): value = 0.052993050777275695, base_value = 0.05295400749413587
		Max relative difference is at index (np.int64(2643), np.int64(0)): value = 3.63995352012823e-18, base_value = 3.467236708214421e-21
	Statistics of the q values greater than 1.0 defined by absolute tolerance: N = 1232
		max = 3.9043283139826577, mean = 0.31018273308305666, std = 0.8999556883939195
		max is at index (np.int64(3337), np.int64(2)), value = 0.052993050777275695, base_value = 0.05295400749413587
	Statistics of the q values greater than 1.0 defined by relative tolerance: N = 0
********************************************************************************
********************************************************************************
Error: /Problem/domain/MeshBodies/mesh/meshLevels/Level0/ElementRegions/elementRegionsGroup/reservoir/elementSubRegions/hexahedra/globalCompDensity
	Arrays of types float64 and float64 have 300 values of which 300 fail both the relative and absolute tests.
		Max absolute difference is at index (np.int64(127), np.int64(1)): value = 990.4965609336606, base_value = 1033.6158879891204
		Max relative difference is at index (np.int64(145), np.int64(0)): value = 9.830636800993778, base_value = 0.00028436378927075074
	Statistics of the q values greater than 1.0 defined by absolute tolerance: N = 45
		max = 2434156.209658291, mean = 235406.16614701573, std = 610313.7952750626
		max is at index (np.int64(127), np.int64(0)), value = 37.21953027905491, base_value = 12.877968182471994
	Statistics of the q values greater than 1.0 defined by relative tolerance: N = 255
		max = 761325.7271238628, mean = 129221.33479600628, std = 174294.33208988397
		max is at index (np.int64(97), np.int64(0)), value = 62.131763000015084, base_value = 53.92141144485521
********************************************************************************
```
the reason is a bug that appears here: https://github.com/GEOS-DEV/GEOS/blob/e601fad71502e0af6b0117af2f8c212ee11ca720/src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp#L886
where `compAmount` does not include the effect of netToGross (don't ask me why).

The issue is addressed in this PR here https://github.com/GEOS-DEV/GEOS/blob/1313a67493f0d45be8a916425d01fcbb89a5eada/src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBase.cpp#L523

**Comment by paveltomin** [comment]
@CusiniM need your review

**Comment by paveltomin** [comment]
@CusiniM please review

**Comment by paveltomin** [comment]
Waiting on code owner review from @CusiniM and/or @ryar9534.

**Review by ryar9534 — APPROVED** [review]

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp
Comment (ryar9534) [inline src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp]: Should this be const or non-const?

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp
Comment (ryar9534) [inline src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: Was this function just not used?

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: yes, not used at all

File: src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp
Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp]: const, thanks

**Comment by paveltomin** [comment]
@CusiniM, @castelletto1, @cssherman, @jhuang2601, @rrsettgast, and/or @wrtobin need your approval for rebaselines

**Comment by paveltomin** [comment]
@CusiniM, @castelletto1, @cssherman, @jhuang2601, @rrsettgast, and/or @wrtobin still need your approval for rebaselines

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp
Comment (CusiniM) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: this is used for the phase field runs I think. @frankfeifan ?

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: it is only implemented in `SinglePhaseHybridFVM` and is not called anywhere, but maybe i am missing something

Comment (CusiniM) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: It's the first part of [this](https://github.com/GEOS-DEV/GEOS/pull/2125/files#diff-5baed9381deba141da8e1689df8266136f85a35a73e53499931634d0148025af) work that was merged. we have not merged the rest yet but that function is needed.

**Review by CusiniM — APPROVED** [review]

File: src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp
Comment (frankfeifan) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: Yes. I will need this function in my [branch](https://github.com/GEOS-DEV/GEOS/pull/2125/files#diff-5baed9381deba141da8e1689df8266136f85a35a73e53499931634d0148025af) and we will work to get my branch merged too. Since this PR has been merged already, I can add this function back to my branch.

Comment (paveltomin) [reply src/coreComponents/physicsSolvers/fluidFlow/SinglePhaseHybridFVM.hpp:167]: it should be restored soon by https://github.com/GEOS-DEV/GEOS/pull/3472
sorry for my misunderstanding

**Pull request merged**
